daphne_worker/src/config.rs

```diff
@@ -329,17 +329,17 @@ impl<D> DaphneWorkerConfig<D> {
             .ok_or_else(|| DapError::fatal("unrecognized task"))
     }
 
-    pub(crate) async fn durable_post<I: Serialize, O: for<'a> Deserialize<'a>>(
+    pub(crate) async fn durable_post<I: Serialize>(
         &self,
         durable_binding: &'static str,
         durable_path: &'static str,
         durable_name: String,
         data: I,
-    ) -> Result<O> {
+    ) -> Result<Response> {
         let namespace = self.durable_object(durable_binding)?;
         let stub = namespace.id_from_name(&durable_name)?.get_stub()?;
-        let mut resp = durable_post!(stub, &durable_path, &data).await?;
-        resp.json().await
+        let resp = durable_post!(stub, &durable_path, &data).await?;
+        Ok(resp)
```

Comment:
Can you explain why this change is needed? It seems to complicate the code below unnecessarily.

Reply:
:( https://github.com/cloudflare/daphne/pull/97#discussion_r954391489

Reply:
Thanks for the feed back.
Apparently, we cannot remove the `let` statement or the Rust compiler will complain.
If we use `resp.json().await`, then we need to explicitly declare what we are expecting in the response `let _response: Vec<Something>`, which makes it very cumbersome when dealing with cases where DO responses are simply empty.
That's why responding with `Ok(resp)` is much simpler, because we can declare `let _response: Vec<Response>` and obtain the result if needed.

I agree that this change makes the code very messy, but I haven't yet found a better way to solve this. Of course, if we can simply get away without declaring a variable in response to `try_join_all()`, then that would be great.

Reply:
Ohhh, I guess what we can do is make sure that each result is equal to the empty string in this case?
```rust
for result in try_join_all!(requests).await? {
    if result != "" {
        // abort
    }
}
```

Reply:
Does this make sense?

Reply:
There is an issue in the implementation here:
In the DO AggregateStore, [some responses](https://github.com/cloudflare/daphne/blob/6228556c7b87a7fe85e414a3186a2511407896f0/daphne_worker/src/durable/aggregate_store.rs#L84) are encoded in JSON and [some](https://github.com/cloudflare/daphne/blob/6228556c7b87a7fe85e414a3186a2511407896f0/daphne_worker/src/durable/aggregate_store.rs#L74) return with an empty response.
In Cloudflare Workers, `from_json` sets the HTTP header to have `application/json` (see [here](https://github.com/cloudflare/workers-rs/blob/main/worker/src/response.rs#L41)) while the empty response does not populate the HTTP header with any content type.

When we run `resp.json().await`, `resp.json()` assumes that `resp` has the HTTP header set to `application/json` and errors out if it doesn't.

We have several ways to deal with responses that do and don't have this content-type:
1. return `Ok(resp)` instead of `resp.json().await` and have the caller decide whether to deserialize, or
2. change AggregateStore so that we always return a JSON body, but just an empty one

Which do you think is better?

Reply:
Thanks for the very clear description of the problem, plus options. I think (2.) is better.

Reply:
Thank you for the suggestion! I've implemented this in the new commit.